### PR TITLE
Biosys 308 record geo filter

### PR DESCRIPTION
**Pull request by serge-gaia**
See https://youtrack.gaiaresources.com.au/youtrack/issue/BIOSYS-308  
and  
https://decbugs.com/view.php?id=7021

### CODE REVIEW

File: biosys/apps/main/tests/api/test_filtering_and_ordering.py [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+
+from django.urls import reverse
+from django.contrib.gis.geos import Polygon, Point
+from rest_framework import status
+
+from main.models import Record, Dataset
+from main.tests.api import helpers
+
+
+class TestJsonSearchAndOrdering(helpers.BaseUserTestCase):
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_filtering_and_ordering.py:11]: This class has been moved to this new test file. You can skip it.

File: biosys/apps/main/tests/api/test_filtering_and_ordering.py [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+            'data__has_key': 'When'
+        }
+        resp = client.get(url, param, format='json')
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        records = resp.json()
+        self.assertEqual(len(records), expected_number)
+
+
+class TestSpatialFiltering(helpers.BaseUserTestCase):
+    """
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_filtering_and_ordering.py:468]: This the test case for the geometry filter

File: biosys/apps/main/tests/api/test_filtering_and_ordering.py [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+
+        self.wa_north_ds = self._create_dataset_and_records_from_rows([
+            ['Species Name', 'When', 'Latitude', 'Longitude', 'Where'],
+            ['Canis lupus', '2018-06-22', -24.882248, 113.662995, 'Carnavon'],
+            ['Quokka', '2018-04-01', -23.142194, 113.772222, 'Coral Bay'],
+            ['Eucalyptus robusta', '2017-11-23', -21.177125, 119.765361, 'Marble Bar'],
+            ['Chubby Bat', '2018-08-23', -17.962075, 122.234554, 'Broome']
+        ])
+
+    def test_get_record_happy(self):
```
Comment (tony-gaia) [inline biosys/apps/main/tests/api/test_filtering_and_ordering.py]: records (with an s)

File: biosys/apps/main/tests/api/test_filtering_and_ordering.py [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+        response = client.get(url, query)
+        self.assertEqual(200, response.status_code)
+        records = response.json()
+        self.assertIsInstance(records, list)
+        self.assertEqual(len(records), len(expected_records))
+        # check location
+        self.assertEqual(
+            sorted([r['data'].get('Where') for r in records]),
+            ['Cottesloe', 'Kings Park', 'Rottnest Island']
+        )
```
Comment (tony-gaia) [inline biosys/apps/main/tests/api/test_filtering_and_ordering.py]: I think this test should have a bounding box that includes both datasets, so as to ensure you only get the records from the dataset you specify.

**Review by tony-gaia — CHANGES_REQUESTED** [review]

File: biosys/apps/main/tests/api/test_filtering_and_ordering.py (hunk 4 above)
Comment (serge-gaia) [reply biosys/apps/main/tests/api/test_filtering_and_ordering.py]: Yep good idea.

**Pull request merged**